iceoryx2-cli/iox2/src/commands.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use colored::*;
+use std::env;
+use std::fs;
+use std::io;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use thiserror::Error;
+
+#[derive(Clone, Debug)]
+struct CommandInfo {
+    name: String,
+    path: PathBuf,
+    is_development: bool,
+}
+
+#[derive(Error, Debug)]
+pub enum ExecutionError {
+    #[error("Command not found: {0}")]
+    NotFound(String),
+    #[error("Execution failed: {0}")]
+    Failed(String),
+}
+
+pub fn list() {
+    println!("{}", "Installed Commands:".bright_green().bold());
+    let mut installed_commands = find();
+    installed_commands.sort_by_key(|command| command.name.clone());
+    installed_commands
+        .iter()
+        .map(|command| {
+            format!(
+                "  {} {}",
+                command.name.bold(),
+                if command.is_development {
+                    "(dev) ".italic()
+                } else {
+                    "".italic()
+                },
+            )
+        })
+        .for_each(|formatted_command| println!("{}", formatted_command));
+}
+
+fn get_development_binaries_dir() -> Result<std::path::PathBuf, io::Error> {
+    let current_exe = env::current_exe()?;
+
+    let build_type = if cfg!(debug_assertions) {
+        "debug"
+    } else {
+        "release"
+    };
+
+    let build_path = current_exe
+        .parent()
+        .and_then(|p| p.parent())
+        .map(|p| p.join(build_type))
+        .ok_or_else(|| {
+            io::Error::new(
+                io::ErrorKind::Other,
+                "Failed to construct path of development dir",
+            )
+        })?;
+
+    Ok(build_path)
+}
+
+fn get_installed_binaries_dirs() -> impl Iterator<Item = std::path::PathBuf> {
```

Comment:
Doesn't this return an iterator of all paths that can contain binaries from the env variable `PATH`? Wouldn't then fit `get_path_env` fit better?!

Reply:
@elfenpiff I understood it as returning an iterator of all "install paths" i.e. those listed in the `PATH` variable, hence the name. 

Would `get_install_path_dirs` be suitable ?